src/DynamoCore/Graph/Nodes/CodeBlockNode.cs

```diff
@@ -827,18 +827,18 @@ internal void SetErrorStatePortData(List<string> inputPortNames, List<int> outpu
 
         private void SetOutputPorts()
         {
-            var allDefs = CodeBlockUtils.GetDefinitionLineIndexMap(codeStatements);
-
-            if (allDefs.Any() == false)
-                return;
-
             // This extension method is used instead because 
             // observableCollection has very odd behavior when cleared - 
             // there is no way to reference the cleared items and so they 
             // cannot be cleaned up properly
             
             // Clear out all the output port models
-            OutPorts.RemoveAll((p) => { return true; });
+            OutPorts.RemoveAll((p) => true);
+            
+            var allDefs = CodeBlockUtils.GetDefinitionLineIndexMap(codeStatements);
+
+            if (allDefs.Any() == false)
+                return;
```

Comment:
ah So always clean the out put ports and re-add them back.  Is there similar pattern we need to worry about input ports or they are already following this?

Reply:
It's already being done for input ports.